external_services/lti.py

```diff
@@ -75,27 +57,55 @@ def __init__(self, service, user, instance, host, title, context_id=None, link_i
             "context_label": course.code,
 
             "launch_presentation_locale": get_language(),
+            "launch_presentation_document_target": "window",
+            "launch_presentation_return_url": request.scheme + '://' + request.get_host() + instance.get_absolute_url(),
 
-            "tool_consumer_instance_guid": host + "/aplus",
+            "tool_consumer_instance_guid": request.get_host() + "/aplus",
```

Comment:
This will need to be refactored to come from django settings in the future or from lti settings. At least in local container environments it would be handy to pass different host for different services (how to call back).

Reply:
tool_consumer_instance_guid is not a callback, though. From the specs:

tool_consumer_instance_guid=lmsng.school.edu
This is a unique identifier for the TC.  A common practice is to use the DNS of the organization or the DNS of the TC instance.  If the organization has multiple TC instances, then the best practice is to prefix the domain name with a locally unique identifier for the TC instance.  In the single-tenancy case, the tool consumer data can be often be derived from the oauth_consumer_key.  In a multi-tenancy case this can be used to differentiate between the multiple tenants within a single installation of a Tool Consumer. This parameter is strongly recommended in systems capable of multi-tenancy.

Reply:
Yep. and thus this parameter shouldn't be generated from client browser (Host header). But I keep it in mind. Should be changed at some point on later date. Actually, I'll create and issue about it.